### PR TITLE
update `gh repo create` docs

### LINKED ISSUE #11931: Update `gh repo create` documentation
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

- [Create a repo](https://docs.github.com/en/get-started/quickstart/create-a-repo)
- [Adding an existing project to GitHub using the command line](https://docs.github.com/en/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line)
- [Creating GitHub CLI extensions](https://docs.github.com/en/github-cli/github-cli/creating-github-cli-extensions)

### What changes are you suggesting?

Update the documentation for `gh repo create` to reflect the new usage, which is set to be merged in the next release of GitHub CLI
Ref: https://github.com/cli/cli/pull/4578, https://github.com/cli/cli/issues/4567 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by meiji163**
### Why:

Closes #11931

### What's being changed:
User guides for using the GitHub CLI to create repositories in the following articles:

- [Create a repo](https://docs.github.com/en/get-started/quickstart/create-a-repo)
- [Adding an existing project to GitHub using the command line](https://docs.github.com/en/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line)
- [Creating GitHub CLI extensions](https://docs.github.com/en/github-cli/github-cli/creating-github-cli-extensions)

Changes to `gh repo create` will be effective once https://github.com/cli/cli/pull/4578 is merged and released 
### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@meiji163 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by meiji163** [comment]
cc/ @skedwards88 @pxrth9

File: content/get-started/quickstart/create-a-repo.md
Comment (skedwards88) [inline content/get-started/quickstart/create-a-repo.md:64]: style guide nit
```suggestion
2. To create a repository for your project, use the `gh repo create` subcommand. When prompted, select **Create a new repository on GitHub from scratch** and enter the name of your new project. If you want your project to belong to an organization instead of to your user account, specify the organization name and project name with `organization-name/project-name`. 
```

File: content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md
Comment (skedwards88) [inline content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md:34]: Should the first step ("In the command line, navigate to the root directory of your project") be omitted now that we are including `-C <em>path/to/repo</em>`?

File: content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md
Comment (skedwards88) [inline content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md:43]: Should we say this instead?:
```suggestion
1. To create a repository for your project on GitHub, use the `gh repo create` subcommand. When prompted, select **Push an existing local repository to GitHub** and enter the desired name for your repository. If you want your project to belong to an organization instead of your user account, specify the organization name and project name with `organization-name/project-name`.
```

**Review by skedwards88 — COMMENTED** [review]
🎉 Thank you for opening this PR! I left some minor comments, but this looks great 💖

File: content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md
Comment (meiji163) [reply content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md:34]: I agree it's more user-friendly without `-C`. I think @pxrth9 wanted to show how the new `--source` flag works with any path, but we can save that for elsewhere

File: content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md
Comment (skedwards88) [inline content/github/importing-your-projects-to-github/importing-source-code-to-github/adding-an-existing-project-to-github-using-the-command-line.md:40]:
```suggestion
   git add . && git commit -m "initial commit"
```

**Review by skedwards88 — DISMISSED** [review]
🎉 This looks great! I'll hold off on merging it until https://github.com/cli/cli/pull/4578 is merged. Changes will go live in ~24 hours after merging.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**
